src/foundry/common/prosemirror/plugin.d.mts

```diff
@@ -19,5 +19,5 @@ declare abstract class ProseMirrorPlugin {
    * @param options - Additional options to pass to the plugin.
    * @virtual
    */
-  static build(schema: Schema, options?: Record<string, unknown>): Plugin;
+  static build(schema: Schema, options?: Record<string, unknown> | undefined): Plugin;
```

Comment:
So `| undefined` in an optional function parameter has no effect. The only difference is with object properties. So you don't need `| undefined` here or in other functions.